soapui/src/main/java/com/eviware/soapui/impl/wsdl/teststeps/assertions/basic/XQueryContainsAssertion.java

```diff
@@ -282,6 +282,8 @@ public int differenceFound(Difference diff) {
                     } catch (ComparisonFailure e) {
                         return Diff.RETURN_ACCEPT_DIFFERENCE;
                     }
+                    
+                    return Diff.RETURN_IGNORE_DIFFERENCE_NODES_IDENTICAL;

```

Comment:
Personally I think it's clearer if you move this line so it comes immediately after the call to assertSimilar(), inside the try block.